test/addons-napi/3_callbacks/test.js

```
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+const addon = require(`./build/${common.buildType}/binding`);
+
+addon.RunCallback(function(msg) {
+  assert.strictEqual(msg, 'hello world');
+});
+
+const global = function() { return this; }.apply();
+
+function testRecv(desiredRecv) {
+  addon.RunCallbackWithRecv(function() {
+    assert.strictEqual(this,
+                       (desiredRecv === undefined ||
+                        desiredRecv === null) ?
+                       global : desiredRecv);
```

These tests do fail, but the exception is swallowed? That’s not good.